src/persist-client/src/internal/encoding.rs

```diff
@@ -845,6 +855,7 @@ impl<T: Timestamp + Lattice + Codec64> RustType<ProtoRollup> for Rollup<T> {
             ts_codec: T::codec_name(),
             diff_codec: self.state.diff_codec.into_proto(),
             last_gc_req: self.state.state.collections.last_gc_req.into_proto(),
+            active_rollup: Some(self.state.state.collections.active_rollup.into_proto()),
```

Comment:
I think this is part of your roundtripping problem... rollups from before your change will have `None` for this field, get the default filled in, then encode as `Some`. One option would be to filter out zeros here, but better IMO to have `StateCollections::active_rollup` be an option itself.